### DISCUSSION TITLE
Troubleshooting TypeError: argument entity must be string, Buffer, or fs.Stats

### DISCUSSION THREAD

**Discussion by esodesod** [coddingtonbear/obsidian-local-rest-api#78] [General]
(Thanks for this awesome plugin, Adam!)

In the last few days, I've been unable to successfully use Local REST API. The plugin and (both) ports are listening as expected but the plugins seem to throw an "internal error", when being used.

If I look in the Developer Tools > Console, I see this, specific for the plugin:

```text
plugin:obsidian-local-rest-api:41524 TypeError: argument entity must be string, Buffer, or fs.Stats
    at etag (plugin:obsidian-local-rest-api:39419:15)
    at generateETag (plugin:obsidian-local-rest-api:40374:16)
    at ServerResponse.send2 [as send] (plugin:obsidian-local-rest-api:43551:20)
    at ServerResponse.json (plugin:obsidian-local-rest-api:43579:19)
    at RequestHandler.returnCannedResponse (plugin:obsidian-local-rest-api:45578:63)
    at RequestHandler.eval (plugin:obsidian-local-rest-api:46188:12)
    at Generator.next (<anonymous>)
    at eval (plugin:obsidian-local-rest-api:79:61)
    at new Promise (<anonymous>)
    at __async (plugin:obsidian-local-rest-api:63:10)
```

If paused on exception, this seems to be the function throwing the error in the console

```js
function etag(entity, options) {
      if (entity == null) {
        throw new TypeError("argument entity is required");
      }
      var isStats = isstats(entity);
      var weak = options && typeof options.weak === "boolean" ? options.weak : isStats;
      if (!isStats && typeof entity !== "string" && !Buffer.isBuffer(entity)) {
        throw new TypeError("argument entity must be string, Buffer, or fs.Stats");
      }
      var tag = isStats ? stattag(entity) : entitytag(entity);
      return weak ? "W/" + tag : tag;
    }
```

I guess there are some "issues" with `etag` (based on my tags/metadata in my files/cache), but I'm looking for a way to detect exactly which file is causing issues, or how to check the entity/values to question. Is this possible to output somehow? Maybe built-in error handling could add some brief information on what file/entity/value is causing the error, as well as the actual error?

Last time something similar happened (back in 2022), I actually detected which file was causing issues via tcpdump, looking at the "regexp", but this time, tcpdump only outputs what I see in the console.

I'll dig around in the Developer Tools and version history for my (recent) notes, to see if I can spot something, but would be lovely if the plugin could somehow include details on what note/metadata to fix. 😀

**Comment by esodesod** [comment]
Small update: I've added `console.log("Entity: " + (entity));` to the "if block" below, while playing around a bit. It seems to output the expected entity filenames, e.g. matching URLs via note recall.

That said, Local Rest API still throws an error, and Obsidian Web results in "ERR":

> Could not connect to Obsidian! Make sure Obsidian is running and that the Obsidian Local REST API plugin is enabled.

If I "comment out" the `throw new TypeError`-line like below, Obsidian Web seems to be working as it normally would (e.g. note recall works again). That said, I guess the TypeError is there for a reason, hence simply a workaround, for now.

```text
if (!isStats && typeof entity !== "string" && !Buffer.isBuffer(entity)) {
	console.log("Entity: " + (entity));
	//throw new TypeError("argument entity must be string, Buffer, or fs.Stats");
  }
```

**Comment by coddingtonbear** [comment]
That's interesting!  I haven't seen this personally -- do you have any instructions I can follow to try to reproduce this on my end?

I'm sure this is obvious, but in case it isn't, that etag code you've commented-out isn't in this library, but a dependency, so just commenting it out isn't an option.  More than likely, though, this'll just be a situation in which some of the types in Obsidian itself just don't quite match what I'm expecting anymore.

**Comment by coddingtonbear** [comment]
& if we could -- could we centralize the conversations about this on this issue here: https://github.com/coddingtonbear/obsidian-local-rest-api/issues/79?
